packages/grafana-ui/src/components/Combobox/Combobox.tsx

```diff
@@ -33,32 +36,79 @@ function itemFilter(inputValue: string) {
   };
 }
 
+function estimateSize() {
+  return 60;
+}
+
 export const Combobox = ({ options, onChange, value, ...restProps }: ComboboxProps) => {
   const [items, setItems] = useState(options);
   const selectedItem = useMemo(() => options.find((option) => option.value === value) || null, [options, value]);
+  const listRef = useRef(null);
+
+  const styles = useStyles2(getStyles);
+
+  const rowVirtualizer = useVirtual({
+    size: items.length,
+    parentRef: listRef,
+    estimateSize,
+    overscan: 2,
+  });
 
   const { getInputProps, getMenuProps, getItemProps, isOpen } = useCombobox({
     items,
     itemToString,
     selectedItem,
+    scrollIntoView: () => {},
     onInputValueChange: ({ inputValue }) => {
       setItems(options.filter(itemFilter(inputValue)));
     },
     onSelectedItemChange: ({ selectedItem }) => onChange(selectedItem),
+    onHighlightedIndexChange: ({ highlightedIndex, type }) => {
+      if (type !== useCombobox.stateChangeTypes.MenuMouseLeave) {
+        rowVirtualizer.scrollToIndex(highlightedIndex);
+      }
+    },
   });
   return (
     <div>
       <Input suffix={<Icon name={isOpen ? 'search' : 'angle-down'} />} {...restProps} {...getInputProps()} />
-      <ul {...getMenuProps()}>
-        {isOpen &&
-          items.map((item, index) => {
-            return (
-              <li key={item.value} {...getItemProps({ item, index })}>
-                {item.label}
-              </li>
-            );
-          })}
+      <ul className={styles.ulMenu} {...getMenuProps({ ref: listRef })}>
+        {isOpen && (
+          <>
+            <li key="total-size" style={{ height: rowVirtualizer.totalSize }} />
```

Comment:
Hmm.... this is a bit odd...

Should/could this instead be a height on the `<ul>` ?

Reply:
Ahhhh, I think I see.

I think, any way, we would want to have a `<div>` that is the dropdown (max-height on this), and inside that a `<ul>` which has it's appropriate height.

Reply:
I agree, it makes more sense.